lib/appsignal/hooks/shoryuken.rb

```diff
@@ -5,16 +5,23 @@ def call(_worker_instance, queue, sqs_msg, body)
         metadata = {
           :queue => queue
         }
-        exclude_keys = [:job_class, :queue_name, :arguments]
-        metadata.merge!(body.reject { |key| exclude_keys.member?(key.to_sym) })
-        metadata.merge!(sqs_msg.attributes)
 
         options = {
-          :class => body["job_class"],
           :method => "perform",
           :metadata => metadata
         }
-        options[:params] = body["arguments"] if body.key?("arguments")
+
+        if body.respond_to?(:reject)
```

Comment:
maybe `body.is_a?(Hash)` is better here. We call more than just `reject` on body